unistore.js

```diff
@@ -51,14 +50,14 @@ export function createStore(state) {
 			return function() {
 				let args = [state];
 				for (let i=0; i<arguments.length; i++) args.push(arguments[i]);
-				let ret = action.apply(this, args);
+				let ret = action.apply(action, args);
 				if (ret!=null) {
-					if (ret.then) ret.then(setState);
-					else setState(ret);
+					if (ret.then) ret.then(setState.bind(action));
```

Comment:
No need for binding here.

Reply:
Ah I see now - what about passing (state, action) to subscribers? Bind is expensive.